### PR TITLE
Archivist (Scholar) Overhaul

### PR DESCRIPTION
**Pull request by AukariBunn**
This update introduces a rework of the archivist, making it the scholar and adding a brand new feature to the game: teaching.

image

Archivist has always been a neglected role, so I believed it must've been made actually useful. The scholar has better skills and more abilities, including multiple spells such as prestidigitation, fetch and message. They are also able to learn new spells, though for the sake of balance they start with 0 spell points.

image

And now the best part... Teaching! The scholar is able to help other player improve their picked skill once, granting this role a unique ability.

image
image

### CODE REVIEW

**Comment by AukariBunn** [comment]
NOTES:

The teaching spell requires its own icon. I already contacted a spriter so I will add the new icon when its done.

I will make some map changes for the scholar after the harbour update is fully merged to avoid conflicts.

File: code/modules/jobs/job_types/roguetown/yeomen/archivist.dm [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
+	Scholars meticulously preserve and organize ancient texts, safeguarding the collective knowledge of the realm for generations to come. \
+	Nobles and Peasants alike often seek the Scholars expertise on matters of history and fact."
 	flag = ARCHIVIST
 	department_flag = YEOMEN
 	faction = "Station"
 	total_positions = 1
 	spawn_positions = 1
-	spells = list(/obj/effect/proc_holder/spell/invoked/projectile/fetch)
-	allowed_races = RACES_TOLERATED_UP
+	allowed_races = RACES_SHUNNED_UP
```
Comment (Gyrandola) [inline code/modules/jobs/job_types/roguetown/yeomen/archivist.dm:11]: nah

File: code/modules/jobs/job_types/roguetown/yeomen/archivist.dm [hunk 2]
```diff
@@ -32,21 +37,111 @@
...
+		H.change_stat("intelligence", 4)
+		if(H.age == AGE_OLD)
+			H.change_stat("speed", -1)
+			H.change_stat("intelligence", 1)
+		H.mind.adjust_spellpoints(-2)
 
-/datum/outfit/job/roguetown/archivist/post_equip(mob/living/carbon/human/H)         ///////////// gave a lil love with a satchel and dagger. virgin ass needs it.
+/datum/outfit/job/roguetown/templar/post_equip(mob/living/carbon/human/H)
 	..()
 	H.virginity = TRUE
```
Comment (Gyrandola) [inline code/modules/jobs/job_types/roguetown/yeomen/archivist.dm:72]: ??

File: code/modules/jobs/job_types/roguetown/yeomen/archivist.dm (hunk 1 above)
Comment (AukariBunn) [reply code/modules/jobs/job_types/roguetown/yeomen/archivist.dm:11]: The scholar is a yeomen and not a courtier so allowing shunned races to play it is more than justified
image
image
image
image
image

Comment (Gyrandola) [reply code/modules/jobs/job_types/roguetown/yeomen/archivist.dm:11]: if he lives in the keep he shall remain tolerated up

Comment (AukariBunn) [reply code/modules/jobs/job_types/roguetown/yeomen/archivist.dm:11]: As i remember the architect lives in the keep too.
image

Comment (AukariBunn) [reply code/modules/jobs/job_types/roguetown/yeomen/archivist.dm:11]: I actually might add them a place outside the keep

**Comment by Gyrandola** [comment]
Testmerging with the agreed condition that the scholar will be moved to the city in a later update.

**Comment by ophaq** [comment]
Don't forget to make them forced as a virgin because they're a nerd. Just kidding, but maybe...

**Comment by AukariBunn** [comment]
A bit about the balance changes

1. Now any skill except arcane can be taught even if you have no novice skill

2. Fixed the bug allowing to train masters (basically could've used to grant roundstart legendary arcane to the magos)

3. There are now two different kinds of learning: if you have no or novice skill, you will get two skill levels, if you have apprentice or a higher skill you will get one skill level. Its much easier to get low level skills so training from novice to journeyman takes roughtly the same effort as training from journeyman to expert.

**Comment by AukariBunn** [comment]
Also the time needed to teach a lesson was increased from 10 seconds to 30 seconds. The teaching spell will also have a 30 second cooldown.

**Comment by john-pawson** [comment]
Hey, i just did noticed that you removed the need for novice to start learning a skill, and now it seems that people are going to the scholar now to completely bypass the need to train blacksmithing until novice, which was the hardest part as intended, since everyone grinding smithing skill and then having no guild smiths because of this is already a very present problem.

Please bring back the need for novice skill.

**Comment by Frienderanian** [comment]
I love this rework but I agree with the other poster here. The Scholar needs to start charging people for teaching or they're just going to be able to teach everyone smithing/artifacer and invalidate entire roles. Maybe it is good to have them need Novice first?

**Comment by Gyrandola** [comment]
Yeah good point. Revert it

**Comment by AukariBunn** [comment]
Another rebalance:

Learning once again require you to have novice skill for the most of the lessons, however, you can learn reading, cooking, sewing and music without any skill.

Also fixed the PQ requirement, back to 5 from 0.

I also just spotted that music has been put twice in the list of the skills that doesnt require you the novice skill, i will release the fix together with the map changes.

**Comment by AukariBunn** [comment]
> I love this rework but I agree with the other poster here. The Scholar needs to start charging people for teaching or they're just going to be able to teach everyone smithing/artifacer and invalidate entire roles. Maybe it is good to have them need Novice first?

The scholar is actually supposed to be charging people for teaching, they are a private tutor. We just dont do that now for the people to become more familiar with the feature as well as for me to get more feedback.

**Comment by AukariBunn** [comment]
Added the academy building near the garrison, the scholar has been moved to there. The bedroom in the keep's archive was deleted.
image
image
image
image

**Review by Gyrandola — APPROVED** [review]

**Pull request merged**